types/index.d.ts

```diff
@@ -1,3 +1,5 @@
+/// <reference path="./PipelineStage.d.ts" />
```

Comment:
While I do like breaking up `index.d.ts` into separate files, is it safe to rely on `reference path`? This means that Mongoose wouldn't work in libs that use [`noResolve`](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#using---noresolve)

Reply:
Interesting, never used noResolve before in my TS projects.

As I read noResolve means that typings like node typings with triple Slash reference won't throw an error if we don't have the node typings installed in our project.

I am currently just on my phone. But we could check this by setting noResolve in tsd configuration. If it throws it is an issue.

Reply:
I checked it. It will break. But it would also break in mongoose 6.2.0 before we merged some of my improvements. So we can safely assume that this is not an actual issue. Also I think before somebody would use noResolve he would use skipLibCheck, which should do what you actually would like to do with noResolve.

If this satisfying your question?

Reply:
@Uzlopak that makes sense, thanks for clarifying. I don't know enough about TypeScript to know whether anyone would actually use `--noResolve`, but it looks like that enough typings in DefinitelyTyped use `reference path` that anyone using noResolve will have a workaround. So I'll merge this :+1: